src/Workspaces/SharedUtilitiesAndExtensions/Workspace/VisualBasic/Extensions/CastExpressionSyntaxExtensions.vb

```diff
@@ -10,21 +10,127 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions
     Friend Module CastExpressionSyntaxExtensions
         <Extension>
         Public Function Uncast(cast As CastExpressionSyntax) As ExpressionSyntax
-            Return Uncast(cast, cast.Expression)
+            Return Uncast(cast, cast.OpenParenToken, cast.Expression, cast.CommaToken, cast.Type, cast.CloseParenToken)
         End Function
 
         <Extension>
         Public Function Uncast(cast As PredefinedCastExpressionSyntax) As ExpressionSyntax
-            Return Uncast(cast, cast.Expression)
+            Return Uncast(cast, cast.OpenParenToken, cast.Expression, Nothing, Nothing, cast.CloseParenToken)
         End Function
 
-        Private Function Uncast(castNode As ExpressionSyntax, innerNode As ExpressionSyntax) As ExpressionSyntax
-            Dim resultNode = innerNode.WithTriviaFrom(castNode)
+        Private Function Uncast(castNode As ExpressionSyntax, openParen As SyntaxToken, innerNode As ExpressionSyntax, commaToken As SyntaxToken, typeNode As TypeSyntax, closeParen As SyntaxToken) As ExpressionSyntax
+
+            Dim leadingTrivia As List(Of SyntaxTrivia) = castNode.GetLeadingTrivia.ToList
```

Comment:
Usea pooled builder

Reply:
Missed pooled builder change, I will have to look into that I am not familiar with it.

Reply:
this would sitll need to be done.  see usages of ArrayBuilder in this project.